src/Jackalope/Transport/DoctrineDBAL/CachedClient.php

```diff
@@ -134,10 +134,23 @@ public function getNode($path)
 
         $cacheKey = "nodes: $path, ".$this->workspaceName;
         if (isset($this->caches['nodes']) && (false !== ($result = $this->caches['nodes']->fetch($cacheKey)))) {
+            if ('' === $result) {
+                throw new ItemNotFoundException("Item $path not found in workspace ".$this->workspaceName);
```

Comment:
also fixed .. though I just copied this from the Client class where we currently do not use sprintf in most places. we should maybe go through it all in a separate PR